optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py

```diff
@@ -0,0 +1,1034 @@
+            ):
+                raise ValueError(
+                    f"`denoising_start`: {self.denoising_start} cannot be larger than or equal to `denoising_end`: "
+                    + f" {self.denoising_end} when using type float."
+                )
+            elif self.denoising_end is not None and denoising_value_valid(self.denoising_end):
+                discrete_timestep_cutoff = int(
+                    round(
+                        self.scheduler.config.num_train_timesteps
+                        - (self.denoising_end * self.scheduler.config.num_train_timesteps)
+                    )
+                )
+                num_inference_steps = len(list(filter(lambda ts: ts >= discrete_timestep_cutoff, timesteps)))
+                timesteps = timesteps[:num_inference_steps]
+
+            # 12.1 Optionally get Guidance Scale Embedding
+            timestep_cond = None
+            if self.unet.config.time_cond_proj_dim is not None:
+                guidance_scale_tensor = torch.tensor(self.guidance_scale - 1).repeat(batch_size)
+                timestep_cond = self.get_guidance_scale_embedding(
+                    guidance_scale_tensor, embedding_dim=self.unet.config.time_cond_proj_dim
+                ).to(device=device, dtype=latents.dtype)
+
+            self._num_timesteps = len(timesteps)
+
+            outputs = {
+                "images": [],
+            }
+            t0 = time.time()
+            t1 = t0
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                latents_batch = latents_batches[0]
+                latents_batches = torch.roll(latents_batches, shifts=-1, dims=0)
+                noise_batch = noise_batches[0]
+                noise_batches = torch.roll(noise_batches, shifts=-1, dims=0)
+                prompt_embeds_batch = prompt_embeds_batches[0]
+                prompt_embeds_batches = torch.roll(prompt_embeds_batches, shifts=-1, dims=0)
+                add_text_embeds_batch = add_text_embeds_batches[0]
+                add_text_embeds_batches = torch.roll(add_text_embeds_batches, shifts=-1, dims=0)
+                add_time_ids_batch = add_time_ids
+                mask_batch = mask_batches[0]
+                mask_batches = torch.roll(mask_batches, shifts=-1, dims=0)
+                if masked_image_latents_batches is not None:
+                    masked_image_latents_batch = masked_image_latents_batches[0]
+                    masked_image_latents_batches = torch.roll(masked_image_latents_batches, shifts=-1, dims=0)
+
+                if image_latents_batches is not None:
+                    image_latents_batch = image_latents_batches[0]
+                    image_latents_batches = torch.roll(image_latents_batches, shifts=-1, dims=0)
+
+                if j == throughput_warmup_steps:
```

Comment:
Please use updated logic for warmup here. See https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L498
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L507
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L601